elide-model-config/src/main/java/com/yahoo/elide/modelconfig/validator/DynamicConfigValidator.java

```diff
@@ -174,57 +175,114 @@ private static void validateInheritance(ElideTableConfig tables, Table table, Se
+                    tables.add(table);
+                    continue;
+                }
+
+                Map<String, Measure> measures = (Map<String, Measure>) getInheritedAttribute(table,
+                        new HashMap<String, Measure>(), (tab, result) -> {
+                            tab.getMeasures().forEach(measure -> {
+                                if (!((Map<String, Measure>) result).containsKey(measure.getName())) {
+                                    ((Map<String, Measure>) result).put(measure.getName(), measure);
+                                }
+                            });
+                            return result;
+                        }, false
+                );
                 table.setMeasures(new ArrayList<Measure>(measures.values()));
 
-                Map<String, Dimension> dimensions = getInheritedDimensions(table, new HashMap<String, Dimension>());
+                Map<String, Dimension> dimensions = (Map<String, Dimension>) getInheritedAttribute(table,
+                        new HashMap<String, Dimension>(), (tab, result) -> {
+                            tab.getDimensions().forEach(dim -> {
+                                if (!((Map<String, Dimension>) result).containsKey(dim.getName())) {
+                                    ((Map<String, Dimension>) result).put(dim.getName(), dim);
+                                }
+                            });
+                            return result;
+                        }, false
+                );
                 table.setDimensions(new ArrayList<Dimension>(dimensions.values()));
 
-                Map<String, Join> joins = getInheritedJoins(table, new HashMap<String, Join>());
+                Map<String, Join> joins = (Map<String, Join>) getInheritedAttribute(table,
+                        new HashMap<String, Join>(), (tab, result) -> {
+                            tab.getJoins().forEach(dim -> {
+                                if (!((Map<String, Join>) result).containsKey(dim.getName())) {
+                                    ((Map<String, Join>) result).put(dim.getName(), dim);
+                                }
+                            });
+                            return result;
+                        }, false
+                );
                 table.setJoins(new ArrayList<Join>(joins.values()));
-            }
-        }
-    }
 
-    private Map<String, Measure> getInheritedMeasures(Table table, Map<String, Measure> measures) {
-        table.getMeasures().forEach(m -> {
-            if (!measures.containsKey(m.getName())) {
-                measures.put(m.getName(), m);
+                String schema = (String) getInheritedAttribute(table, null,
+                        (tab, result) -> {
+                            return tab.getSchema();
+                        }, true
+                );
+                table.setSchema(schema);
+
+                String dbConnectionName = (String) getInheritedAttribute(table, null,
```

Comment:
These should all be separate functions.  Having embedded lambdas all together like this in one function makes this hard to read.

Reply:
null as second parameter seems like the abstraction to commonize this code is not working.